tests/integration.rs

```diff
@@ -646,3 +649,65 @@ fn test_call_blocking() -> Result<()> {
 
     Ok(())
 }
+
+#[test]
+fn test_open_key_with_flags() -> Result<()> {
+    let port: u16 = 6501;
+    let _guards = vec![start_redis_server_with_module("open_key_with_flags", port)
+        .with_context(|| "failed to start redis server")?];
+    let mut con =
+        get_redis_connection(port).with_context(|| "failed to connect to redis server")?;
+
+    // Avoid active expriation
+    redis::cmd("DEBUG")
+        .arg(&["SET-ACTIVE-EXPIRE", "0"])
+        .query(&mut con)
+        .with_context(|| "failed to run DEBUG SET-ACTIVE-EXPIRE")?;
+
+    for read_write in [true, false].iter() {
+        redis::cmd("set")
+            .arg(&["x", "1"])
+            .query(&mut con)
+            .with_context(|| "failed to run string.set")?;
+
+        // Set experition time to 1 second.
+        redis::cmd("expire")
+            .arg(&["x", "1"])
+            .query(&mut con)
+            .with_context(|| "failed to run expire")?;
+
+        // Sleep for 2 seconds, ensure expiration time has passed.
+        thread::sleep(Duration::from_secs(2));
+
+        let cmd = if *read_write {
+            "open_key_with_flags.write"
+        } else {
+            "open_key_with_flags.read"
+        };
+
+        // Open key as read only or ReadWrite with NOEFFECTS flag.
+        let res = redis::cmd(cmd).arg(&["x"]).query(&mut con);
+        assert_eq!(res, Ok(()));
+
+        // Get the number of expired keys.
+        let stats: String = redis::cmd("info").arg(&["stats"]).query(&mut con)?;
+
+        // Find the number of expired keys, x,  according to the substring "expired_keys:{x}"
+        let expired_keys = stats
+            .match_indices("expired_keys:")
+            .next()
+            .map(|(i, _)| &stats[i..i + "expired_keys:".len() + 1])
+            .and_then(|s| s.split(':').nth(1))
+            .and_then(|s| s.parse::<i32>().ok())
+            .unwrap_or(-1);
```

Comment:
@DvirDukhan Consider using a regex to capture the number of expired keys
(also need to make `utils` module public)
```suggestion
        use redis_module::utils::get_regexp_captures;
        let expired_keys = get_regexp_captures(stats.as_str(), r"\bexpired_keys:([0-9]+)\b")
            .unwrap()[0]
            .parse::<i32>()
            .unwrap_or(-1);
```